src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java

```diff
@@ -278,8 +280,70 @@ public ModelingSubmissionResource(ModelingSubmissionService modelingSubmissionSe
     }
 
     /**
-     * Removes sensitive information (e.g. example solution) from the exercise. This should be called before sending an exercise to the client for a student. IMPORTANT: Do not call
-     * this method from a transactional context as this would remove the sensitive information also from the entity in the database without explicitly saving it.
+     * Returns the submission with data needed for the modeling editor, which includes the participation, the model and the result (if the assessment was already submitted).
+     *
+     * @param participationId the participationId for which to find the data for the modeling editor
+     * @return the ResponseEntity with json as body
+     */
+    @GetMapping("/modeling-editor/{participationId}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    public ResponseEntity<ModelingSubmission> getSubmissionForModelingEditor(@PathVariable Long participationId) {
+        Participation participation = participationService.findOneWithEagerSubmissionsAndResults(participationId);
+        if (participation == null) {
+            return ResponseEntity.badRequest()
```

Comment:
This should not be a bad Request (400), but a not found (404).
A 400 would be the case that somebody uses the endpoint without providing the participationId.

Reply:
You are correct, I changed it.